wg-node-lifecycle/charter.md

```diff
@@ -0,0 +1,160 @@
+# WG Node Lifecycle Charter
+
+This charter adheres to the conventions described in the [Kubernetes Charter README] and uses
+the Roles and Organization Management outlined in [wg-governance].
+
+[Kubernetes Charter README]: /committee-steering/governance/README.md
+
+## Scope
+
+The Kubernetes ecosystem currently faces challenges in node maintenance scenarios, with multiple
+projects independently addressing similar issues. The goal of this working group is to develop
+unified APIs that the entire ecosystem can depend on, reducing the maintenance burden across
+projects and addressing scenarios that impede node drain or cause improper pod termination. Our
+objective is to create easily configurable, out-of-the-box solutions that seamlessly integrate with
+existing APIs and behaviors. We will strive to make these solutions minimalistic and extensible to
+support advanced use cases across the ecosystem.
+
+To properly solve the node drain, we must first understand the node lifecycle. This includes
+provisioning/sunsetting of the nodes, PodDisruptionBudgets, API-initiated eviction and node
+shutdown. This then impacts both the node and pod autoscaling, de/scheduling, load balancing, and
+the applications running in the cluster. All of these areas have issues and would benefit from a
+unified approach.
+
+### In scope
+
+- Explore a unified way of draining the nodes and managing node maintenance by introducing new APIs
+  and extending the current ones. This includes exploring extension to or interactions with the Node
+  object.
+- Analyze the node lifecycle, the Node API, and possible interactions. We want to explore augmenting
+  the Node API to expose additional state or status in order to coalesce other core Kubernetes and
+  community APIs around node lifecycle management.
+- Improve the disruption model that is currently implemented by API-initiated Eviction API and PDBs.
+  Improve the descheduling, availability and migration capabilities of today's application
+  workloads. Also explore the interactions with other eviction mechanisms.
+- Improve the Graceful/Non-Graceful Node Shutdown and consider how this affects the node lifecycle.
+  To graduate the [Graceful Node Shutdown](https://github.com/kubernetes/enhancements/issues/2000)
+  feature to GA and resolve the associated node shutdown issues.
+- Improve the scheduling and pod/node autoscaling to take into account ongoing node maintenance and
+  the new disruption model/evictions. This includes balancing of the pods according to scheduling
+  constraints. 
+- Consider improving the pod lifecycle of DaemonSets and Static pods during a node maintenance.
+- Explore the cloud provider use cases and how they can hook in into the node lifecycle. So that the
+  users can use the same APIs or configurations across the board.
+- Migrate users of the eviction based kubectl-like drain (kubectl, cluster autoscaler, karpenter,
+  ...) and other scenarios to use the new approach.
```

Comment:
nit: Make it more clear that when you say "new approach" you're referring to "a unified way of draining the nodes" you mentioned in the first bullet point. Either move this sentence up or clarify it here.